### ISSUE TITLE
Support for deferring a response?

### ISSUE THREAD

**Issue by GP-95** [discordcf/discordcf#35]
Discord expects a response within 3 seconds, otherwise it fails fails the interaction. 

Some other packages support deferring the response while processing is going on, and immediately sends a preliminary response to stop Discord from failing the interaction.

Does this package support such a feature? I could not find it in the source or documentation.

**Comment by Inrixia** [comment]
Yes it does. See https://github.com/Inrixia/LostMokoko/blob/master/src/reactMentions.ts for an example.

I should work on adding that to our examples

**Comment by alvgaona** [comment]
Hey @GP-95, thanks for taking a look at our project. We really appreciate your question.

As @Inrixia suggested, it is supported but you'd have to do some additional work to build the webhook endpoints and so on. You can do it like that for the moment, but I'm actually working on providing a better API so your commands don't need to be aware of the low-level details, such as the actual URL for sending the deferred message.

**Comment by alvgaona** [comment]
@GP-95 is just a preview of the API that I'll add, potentially. It's subject to changes though.

```typescript
export const deferredCommand: Command = [
  {
    name: 'deferred',
    description: 'A deferred message',
  },
  async (ctx: Context): Promise<APIInteractionResponse> => {
    ctx.defer(async () => {
      // Do stuff that take more tan 3 seconds.
      ctx.followup.reply({
        content: "Message was deferred",
      })
    })

    return {
      type: InteractionResponseType.DeferredChannelMessageWithSource,
      data: {
        flags: MessageFlags.Ephemeral,
      },
    };
  },
];
```

I'm trying to abstract the users away from the low-level details so it's easy to use. Hopefully, this API will be documented as well as it brings to play quite a few changes to the current state of the project.

**Comment by alvgaona** [comment]
Feature merged and will be released in `0.1.0`.
